mbuild/recipes/silica_interface.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import division
+
+import mbuild as mb
+import networkx as nx
+import numpy as np
+
+from random import randint, choice
+from math import ceil
+
+
+class SilicaInterface(mb.Compound):
+    """ A recipe for creating an interface from bulk silica.
+
+    Carves silica interface from bulk, adjusts to desired surface
+    hydoxyl density by creating Si-O-Si bridges, and yields a 2:1
+    Si:O ratio (excluding surface binding sites)
+
+    Parameters
+    ----------
+    bulk_silica : mb.Compound
+        Bulk silica from which to cleave an interface
+    tile_x : int, optional, default=1
+        Number of times to replicate bulk silica in x-direction
+    tile_y : int, optional, default=1
+        Number of times to replicate bulk silica in y-direction
+    thickness : float, optional, default=1.0
+        Thickness of the interface (in nm)
+    oh_density : float, optional, default=5.0
+        Desired density of reactive surface sites (sites/nm^2)
+
+    """
+
+    def __init__(self, bulk_silica, tile_x=1, tile_y=1, thickness=1.0, oh_density=5.0):
+        super(SilicaInterface, self).__init__()
+
+        _O_buffer = 0.275
+
+        self._cleave_interface(bulk_silica, tile_x, tile_y, thickness, _O_buffer)
+        self.generate_bonds(name_a='Si', name_b='O', dmin=0.0, dmax=0.20419)
+        self._strip_stray_atoms()
+        self._bridge_dangling_Os(oh_density, thickness)
+        self._identify_surface_sites(thickness)
+        self._adjust_stoichiometry(_O_buffer)
+
+    def _cleave_interface(self, bulk_silica, tile_x, tile_y, thickness, O_buffer):
+        """ Carve interface from bulk silica, include a buffer of O's above and
+            below the surface to ensure the interface is coated.
+        """
+
+        tile_z = int(ceil((thickness + 2*O_buffer) / bulk_silica.periodicity[2]))
+        bulk = mb.TiledCompound(bulk_silica, n_tiles=(tile_x, tile_y, tile_z))
+
+        interface = mb.Compound(periodicity=(bulk.periodicity[0],
+                                             bulk.periodicity[1],
+                                             0.0))
+        for i, particle in enumerate(bulk.particles()):
+            if ((particle.name == 'Si' and O_buffer < particle.pos[2] < (thickness + O_buffer)) or 
+                    (particle.name == 'O' and particle.pos[2] < (thickness + 2*O_buffer))):
+                interface_particle = mb.Compound(name=particle.name, pos=particle.pos)
+                interface.add(interface_particle, particle.name + "_{}".format(i))
+        self.add(interface)
+
+    def _strip_stray_atoms(self):
+        """ Remove stray atoms and surface pieces """
+
+        major_component = max(nx.connected_components(self.bond_graph), key=len)
+        for atom in list(self.particles()):
+            if not self.bond_graph.has_node(atom):
```

Comment:
I think this might be unnecessary since the next check is to see `if atom not in major_component` which should serve the same purpose.